homeassistant/components/openuv/binary_sensor.py

```diff
@@ -27,72 +27,44 @@ async def async_setup_entry(hass, entry, async_add_entities):
     binary_sensors = []
     for kind, attrs in BINARY_SENSORS.items():
         name, icon = attrs
-        binary_sensors.append(
-            OpenUvBinarySensor(openuv, kind, name, icon, entry.entry_id)
-        )
+        binary_sensors.append(OpenUvBinarySensor(openuv, kind, name, icon))
 
     async_add_entities(binary_sensors, True)
 
 
 class OpenUvBinarySensor(OpenUvEntity, BinarySensorEntity):
     """Define a binary sensor for OpenUV."""
 
-    def __init__(self, openuv, sensor_type, name, icon, entry_id):
+    def __init__(self, openuv, sensor_type, name, icon):
         """Initialize the sensor."""
-        super().__init__(openuv)
-
-        self._async_unsub_dispatcher_connect = None
-        self._entry_id = entry_id
-        self._icon = icon
-        self._latitude = openuv.client.latitude
-        self._longitude = openuv.client.longitude
-        self._name = name
-        self._sensor_type = sensor_type
-        self._state = None
-
-    @property
-    def icon(self):
-        """Return the icon."""
-        return self._icon
-
-    @property
-    def is_on(self):
-        """Return the status of the sensor."""
-        return self._state
-
-    @property
-    def should_poll(self):
-        """Disable polling."""
-        return False
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique, Home Assistant friendly identifier for this entity."""
-        return f"{self._latitude}_{self._longitude}_{self._sensor_type}"
+        super().__init__(openuv, sensor_type)
+
+        self._attr_icon = icon
+        self._attr_name = name
 
     @callback
     def update_from_latest_data(self):
         """Update the state."""
         data = self.openuv.data[DATA_PROTECTION_WINDOW]
 
         if not data:
-            self._available = False
+            self._attr_available = False
             return
 
-        self._available = True
+        self._attr_available = True
 
         for key in ("from_time", "to_time", "from_uv", "to_uv"):
             if not data.get(key):
                 LOGGER.info("Skipping update due to missing data: %s", key)
                 return
 
         if self._sensor_type == TYPE_PROTECTION_WINDOW:
-            self._state = (
+            self._attr_state = (
```

Comment:
Binary sensors should use `self._attr_is_on`, I believe this entity isn't working at this point.